app/models/concerns/authorisable.rb

```diff
@@ -0,0 +1,46 @@
+module Authorisable
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :appropriate_body_roles
+    has_many :appropriate_bodies, through: :appropriate_body_roles
+    has_many :delivery_partner_roles
+    has_many :delivery_partners, through: :delivery_partner_roles
+    has_many :lead_provider_roles
+    has_many :lead_providers, through: :lead_provider_roles
+    has_many :school_roles
+    has_many :schools, through: :school_roles
+    has_many :dfe_roles
+  end
+
+  def can_access?(model)
+    return true if dfe_user?
+    return AppropriateBodyRole.where(user: self, appropriate_body: model).exists? if model.is_a?(AppropriateBody)
+    return DeliveryPartnerRole.where(user: self, delivery_partner: model).exists? if model.is_a?(DeliveryPartner)
+    return LeadProviderRole.where(user: self, lead_provider: model).exists? if model.is_a?(LeadProvider)
+    return SchoolRole.where(user: self, school: model).exists? if model.is_a?(School)
+
+    false
+  end
```

Comment:
Much easier to follow. Would a `case` statement be even clearer?

```suggestion
  def can_access?(model)
    case model
    when AppropriateBody
       AppropriateBodyRole.where(user: self, appropriate_body: model).exists?
    when DeliveryPartner
       DeliveryPartnerRole.where(user: self, delivery_partner: model).exists?
    when LeadProvider
      LeadProviderRole.where(user: self, lead_provider: model).exists?
    when School
       SchoolRole.where(user: self, school: model).exists? 
    else
      dfe_user?
  end
```

Reply:
Oops sorry my indentation was a bit skewiff in that suggestion 🙈